content/automate/ManageIQ/PhysicalInfrastructure/CiscoIntersight/GetMethods/GetServers.class/__methods__/get_servers.rb

```diff
@@ -0,0 +1,25 @@
+# Description: get unassigned servers
+$evm.log("info", 'selected_ems for servers')
+
+dialog_field = $evm.object
+dialog_field["sort_by"] = "value"
+dialog_field["sort_order"] = "ascending"
+dialog_field["data_type"] = "string"
+dialog_field["required"] = true
+
+selected_ems_id = $evm.root.attributes['dialog_ems']
+selected_ems = $evm.vmdb(:ext_management_system).where(:id=>selected_ems_id).first
+$evm.log("info", 'selected_ems for servers')
+$evm.log("info", selected_ems.name)
+
+servers = selected_ems.physical_servers
+profiles = selected_ems.physical_server_profiles
+profiles_with_server = profiles.reject { |p| p["assigned_server_id"].nil? }
+servers_with_profile = profiles_with_server.pluck("assigned_server_id")
+
+servers_f = servers.select { |s| !s["name"].nil? && !s.id.in?(servers_with_profile) }
```

Comment:
What is the `f` in `servers_f` ?

Reply:
I set it to appropriate name- commission_servers_without_profile.
it gets the servers that have names and don't assign to profiles.